### PR TITLE
ieee802154 nrf5 callout timeouts

### PR DESCRIPTION
**Pull request by ankuns**
This PR brings fixes allowing to detect crashes of nRF5340 network core running nRF 802.15.4 Radio Driver.
Detection is done in zephyr's ieee802154_nrf5.c shim, see this PR https://github.com/nrfconnect/sdk-zephyr/pull/1199

Also cherry-picked commit https://github.com/nrfconnect/sdk-nrf/commit/5a62e5e37e9ac28242fad8dde52713252e3d1bcc to ensure than `nrf_802154_serialization_error` never returns.

Samples now use CONFIG_RESET_ON_FATAL_ERROR=n for multiprotocol_rpmsg child image (for nRF5340 network core).
A silent reset of network core is detrimental, because the transmission can success, but can be performed on default channel (11) instead of desired, with default power (0dBm) instead of desired. The state of network core becomes other than the app core believes it is.

The goal is to reset **whole nRF5340 SoC** when network core becomes unresponsive (due to a crash).
If the network core crashes, it will hang instead of silent reset.
When the network core crashes during a call to any nrf_802154_xxxxx there will be serialization error with LOG_ERR "no response within timeout 500" (however that's no change, just note it), ending with k_oops on app core leading to reset of the SoC.
When the network core crashes during a transmission procedure, cca or energy detection procedure (the operations starting with some nrf_802154_xxxxx but ending with callouts like nrf_802154_transmitted/nrf_802154_transmit_failed), there will be k_oops on app core leading to reset of the SoC. - that's new behavior. The timeout is Kconfig-uralbe, and can be turned off to keep previous behavior.
When the network core crashes during an listening (with no other activity on 802.15.4 from an application) and frames simply cease to be received, the ieee802154_nrf5.c shim will read current time (with nrf_802154_time_get) once every 60s. This will cause serialization error with LOG_ERR "no response within timeout 500" and k_opps on app core leading to reset of the SoC. - that's new behavior.

### CODE REVIEW

**Review by jciupis — APPROVED** [review]

File: applications/matter_weather_station/configuration/thingy53_nrf5340_cpuapp/child_image/multiprotocol_rpmsg/prj_release.conf [hunk 1]
```diff
@@ -17,4 +17,4 @@ CONFIG_UART_CONSOLE=n
 # Workaround FEM issue seen on older Thingy:53 revisions
 CONFIG_MPSL_FEM_NRF21540_TX_GAIN_DB_POUTA=10
 
-CONFIG_RESET_ON_FATAL_ERROR=y
```
Comment (Damian-Nordic) [inline applications/matter_weather_station/configuration/thingy53_nrf5340_cpuapp/child_image/multiprotocol_rpmsg/prj_release.conf:20]: Can you explain how this all will work if the network core does not reset on failure (e.g. in the PR description)?

Comment (ankuns) [reply applications/matter_weather_station/configuration/thingy53_nrf5340_cpuapp/child_image/multiprotocol_rpmsg/prj_release.conf:20]: @Damian-Nordic please find updated PR description

Comment (Damian-Nordic) [reply applications/matter_weather_station/configuration/thingy53_nrf5340_cpuapp/child_image/multiprotocol_rpmsg/prj_release.conf:20]: Thanks!

**Review by Damian-Nordic — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Review by doublemis1 — APPROVED** [review]

File: west.yml [hunk 2]
```diff
@@ -59,7 +59,7 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: v3.3.99-ncs1
+      revision: pull/1199/head
```
Comment (rlubos) [inline west.yml:62]: `2fb648399bf66ed3cfc978c8f5803f0ba75caf93`

Comment (ankuns) [reply west.yml]: done

**Review by rlubos — APPROVED** [review]

**Pull request merged**